src/app/workspace/workspace.service.ts

```diff
@@ -114,4 +115,46 @@ export class WorkspaceService {
     const user = await this.userRepository.findOneBy({ email });
     return { userExists: user ? true : false };
   }
+  
+  public async createWorkspace(createWorkspaceDto: CreateWorkspaceDto, user: User ):
+  Promise<{projectName: string, settings: string, ownerId: number}> {
+    const workspace = { ...createWorkspaceDto, ownerId: user.id};
+    // const workspace = this.workspaceRepository.create(createWorkspaceDto);
+    return this.workspaceRepository.save(workspace);
```

Comment:
Here you probably want to await this save method and why return type is't Promise<Workspace>?